ui/src/views/PulseMessages/index.jsx

```diff
@@ -267,75 +242,16 @@ export default class PulseMessages extends Component {
           </HelpView>
         }>
         <Fragment>
-          <ErrorPanel fixed error={error} />
-          <div className={classes.inputWrapper}>
-            <List className={classes.inputList}>
-              <ListItem>
-                <ListItemText
-                  primary={
-                    <TextField
-                      required
-                      label="Pulse Exchange"
-                      name="pulseExchange"
-                      placeholder="exchange/<username>/some-exchange-name"
-                      onChange={this.handleInputChange}
-                      fullWidth
-                      value={pulseExchange}
-                    />
-                  }
-                />
-              </ListItem>
-              <ListItem>
-                <ListItemText
-                  primary={
-                    <TextField
-                      required
-                      label="Routing Key Pattern"
-                      placeholder="#.some-interesting-key.#"
-                      name="pattern"
-                      onChange={this.handleInputChange}
-                      fullWidth
-                      value={pattern}
-                    />
-                  }
-                />
-              </ListItem>
-            </List>
-            <Tooltip title="Add Binding">
-              <IconButton
-                className={classNames(classes.iconButton, classes.plusIcon)}
-                onClick={this.handleAddBinding}>
-                <PlusIcon />
-              </IconButton>
-            </Tooltip>
-          </div>
+          <ErrorPanel error={error} />
+          <PulseBindings
+            bindings={bindings}
+            onBindingAdd={this.handleAddBinding}
+            onBindingRemove={this.handleDeleteBinding}
+            onChange={this.handleInputChange}
```

Comment:
If we change `onChange` to `onPulseExchangeChange` and `onRoutingKeyPatternChange`, then we won't need to do the `patternName` hack anymore. I think it would be nice if we can do that.

Reply:
I think this will not help, as main problem here is in HookForm, for pattern we need key to be `rountingKeyPattern` and in PulseMessages we need key to be `pattern`.

Does both the components use same key for pattern? As PulseMessages was using `pattern` as key before my changes. If we have same key for patterns in both the component we would not need any of the above changes.

Reply:
`onRoutingKeyPatternChange` would send the updated value and then we would be able to properly set the state in each form. Something like:

```jsx
// in HookForm
handleRoutingKeyPatternChange = ({ target: { value } }) => {
  this.setState({ routingKeyPattern: value });
}

// ...

<PulseBinding onRoutingKeyPatternChange={this.handleRoutingKeyPatternChange} ... />
```

```jsx
// in PulseMessages
handleRoutingKeyPatternChange = ({ target: { value } }) => {
  this.setState({ pattern: value });
}

// ...

<PulseBinding onRoutingKeyPatternChange={this.handleRoutingKeyPatternChange} ... />
```
Thoughts?

Reply:
We won't need to use `handleInputChange` anymore.

Reply:
> `onRoutingKeyPatternChange` would send the updated value and then we would be able to properly set the state in each form. Something like:
> 
> ```js-jsx
> // in HookForm
> handleRoutingKeyPatternChange = ({ target: { value } }) => {
>   this.setState({ routingKeyPattern: value });
> }
> 
> // ...
> 
> <PulseBinding onRoutingKeyPatternChange={this.handleRoutingKeyPatternChange} ... />
> ```
> 
> ```js-jsx
> // in PulseMessages
> handleRoutingKeyPatternChange = ({ target: { value } }) => {
>   this.setState({ pattern: value });
> }
> 
> // ...
> 
> <PulseBinding onRoutingKeyPatternChange={this.handleRoutingKeyPatternChange} ... />
> ```
> 
> Thoughts?

This makes sense. Let me update the PR with the changes